src/main/java/com/facebook/openwifirrm/modules/ModelerUtils.java

```diff
@@ -381,4 +387,161 @@ public static Map<String, Map<String, WifiScanEntry>> getAggregatedWifiScans(
 		}
 		return aggregatedWifiScans;
 	}
+
+	/**
+	 * This method converts the input State info to an AggregatedState
+	 * and adds it to the bssidToAggregatedStates map. If the bssid&station
+	 * of the input State does not exist in the map, create a new
+	 * AggregatedState list. If the bssid&station of the input State exists,
+	 * then convert State to AggregatedState and check if there exits an
+	 * AggregatedState of the same radio. If there does, append the value
+	 * of aggregation field to the existing AggregatedState, if not, create
+	 * a new AggregatedState and add it to the list.
+	 *
+	 * @param bssidToAggregatedStates map from bssid&station to a list of AggregatedState.
+	 * @param state the state that is to be added.
+	 */
+	static void addStateToAggregation(
+		Map<String, List<AggregatedState>> bssidToAggregatedStates,
+		State state
+	) {
+		for (Interface stateInterface : state.interfaces) {
+			if (stateInterface.ssids == null) {
+				continue;
+			}
+			for (SSID ssid : stateInterface.ssids) {
+				Map<String, Integer> radioInfo = new HashMap<>();
+				radioInfo.put("channel", ssid.radio.get("channel").getAsInt());
+				radioInfo.put(
+					"channel_width",
+					ssid.radio.get("channel_width").getAsInt()
+				);
+				radioInfo
+					.put("tx_power", ssid.radio.get("tx_power").getAsInt());
+
+				for (Association association : ssid.associations) {
+					if (association == null) {
+						continue;
+					}
+					String key = String.format(
+						"bssid: %s, station: %s",
+						association.bssid,
+						association.station
+					);
+					List<AggregatedState> aggregatedStates =
+						bssidToAggregatedStates
+							.computeIfAbsent(key, k -> new ArrayList<>());
+					AggregatedState aggState =
+						new AggregatedState(association, radioInfo);
+
+					boolean toBeAggregated = false;
```

Comment:
Maybe this is just a personal preference/misunderstanding - to me, it makes more sense to call this something like `canBeMerged` or `addedToExistingAggregateState` or something like that?